### PR TITLE
Revamp librispeech dataset

### PR DESCRIPTION
**Pull request by ejguan**
This PR needs https://github.com/pytorch/pytorch/pull/72260 to be landed into nightly build.

cc: @NivekT @nateanl

### CODE REVIEW

**Comment by ejguan** [comment]
My PR in PyTorch didn't make it into nightly release. Have to wait another day. So annoying.

File: examples/audio/librispeech.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        waveform,
+        sample_rate,
+        transcript,
+        int(speaker_id),
+        int(chapter_id),
+        int(utterance_id),
+    )
+
+
+def LibriSpeech(root: Union[str, Path], url: str = URL, folder_in_archive: str = FOLDER_IN_ARCHIVE):
```
Comment (nateanl) [inline examples/audio/librispeech.py:58]: should it be a datapipe class instead of a method?

Comment (ejguan) [reply examples/audio/librispeech.py:58]: We try to use modular DataPipe to build pipeline (a sequence of DataPipe). So, I am using a function to represent how you want to create the pipeline.
The way you build your Dataset is up to domain. For example, TorchText is using method to return DataPipe https://github.com/pytorch/text/tree/main/torchtext/datasets. But, for TorchVision, they create another Wrapper https://github.com/pytorch/vision/tree/main/torchvision/prototype/datasets/_builtin and create a common API to fetch data pipeline `_make_datapipe`

File: examples/audio/librispeech.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    cache_compressed_dp = HttpReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+
+    # Cache decompressed archive
+    cache_decompressed_dp = cache_compressed_dp.on_disk_cache(filepath_fn=lambda x: archive_path)
+    cache_decompressed_dp = FileOpener(cache_decompressed_dp, mode="b").read_from_tar()
+    cache_decompressed_dp = cache_decompressed_dp.end_caching(
+        mode="wb", filepath_fn=lambda filepath: os.path.join(archive_path, os.path.basename(filepath))
+    )
+
+    audio_dp = cache_decompressed_dp.filter(lambda f: f.endswith(AUDIO_EXT))
```
Comment (ejguan) [inline examples/audio/librispeech.py:96]: I plan to add bucketize here.

**Comment by ejguan** [comment]
Closing this PR as I change to use ghstack to submit a stack.

**Pull request closed without merging**